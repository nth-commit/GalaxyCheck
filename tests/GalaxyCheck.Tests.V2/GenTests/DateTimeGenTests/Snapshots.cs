using GalaxyCheck;
using Snapshooter;
using Snapshooter.Xunit;
using System.Linq;
using Xunit;

namespace Tests.V2.GenTests.DateTimeGenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            foreach (var seed in seeds)
            {
                var sample = Gen
                    .DateTime()
                    .Select(x => x.ToString("s"))
                    .Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            foreach (var seed in seeds)
            {
                var sample = Gen.DateTime().RenderOneTraversal(
                    seed: seed,
                    size: 75,
                    renderer: x => x.ToString("s"));

                var nameExtension = string.Join("_", new[]
                {
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
