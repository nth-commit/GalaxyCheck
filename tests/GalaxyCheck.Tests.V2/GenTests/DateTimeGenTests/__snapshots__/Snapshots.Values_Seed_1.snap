[
  "0001-01-01T00:00:00",
  "0001-01-02T00:00:01",
  "0002-01-01T00:00:01",
  "0001-02-02T00:00:01",
  "0002-01-01T00:00:01",
  "0002-02-01T00:00:00",
  "0002-02-02T00:00:00",
  "0002-02-02T00:00:01",
  "0001-01-01T00:00:01",
  "0002-02-01T00:00:00",
  "0002-02-02T00:00:00",
  "0002-01-02T00:00:01",
  "0001-02-02T00:00:00",
  "0001-01-01T00:00:01",
  "0002-01-01T00:00:00",
  "0001-01-02T00:00:00",
  "0001-02-02T00:00:00",
  "0002-01-01T00:00:03",
  "0002-01-02T00:00:03",
  "0002-02-01T00:00:02",
  "0003-04-03T00:00:03",
  "0004-01-02T00:00:01",
  "0004-06-05T00:00:04",
  "0006-06-06T00:00:05",
  "0006-01-04T00:00:00",
  "0001-02-03T00:00:02",
  "0002-02-01T00:00:01",
  "0006-01-05T00:00:03",
  "0006-02-06T00:00:02",
  "0002-05-04T00:00:05",
  "0004-02-05T00:00:04",
  "0002-06-06T00:00:04",
  "0006-05-02T00:00:05",
  "0001-01-03T00:00:02",
  "0001-01-06T00:00:03",
  "0006-03-02T00:00:01",
  "0001-02-01T00:00:03",
  "0002-04-01T00:00:10",
  "0005-02-05T00:00:00",
  "0004-04-03T00:00:01",
  "0004-09-07T00:00:05",
  "0006-11-09T00:00:10",
  "0004-04-10T00:00:05",
  "0008-06-04T00:00:08",
  "0005-07-06T00:00:03",
  "0009-03-01T00:00:08",
  "0002-01-10T00:00:10",
  "0002-10-03T00:00:04",
  "0004-11-09T00:00:09",
  "0009-02-08T00:00:09",
  "0007-06-08T00:00:52",
  "0008-08-01T00:00:24",
  "0006-09-09T00:01:32",
  "0011-02-11T00:01:05",
  "0002-03-10T00:01:14",
  "0005-03-07T00:00:39",
  "0003-06-30T00:01:19",
  "0006-03-05T00:00:21",
  "0008-07-04T23:59:59",
  "0008-12-02T00:00:19",
  "0045-04-22T00:00:34",
  "0058-06-10T00:00:40",
  "0079-01-31T00:00:52",
  "9999-09-16T00:00:57",
  "0032-07-04T00:00:01",
  "9999-11-26T00:01:39",
  "0058-11-28T00:01:40",
  "0070-01-02T00:15:46",
  "0065-11-26T01:00:51",
  "0032-02-25T02:05:17",
  "9999-12-31T02:05:43",
  "0052-06-30T01:25:45",
  "0083-04-30T02:41:55",
  "0034-01-21T01:14:12",
  "0101-03-27T23:59:59",
  "9999-12-01T02:01:03",
  "0013-09-28T01:57:15",
  "0075-09-01T02:24:23",
  "9999-07-10T02:34:23",
  "0041-05-02T00:51:37",
  "3195-07-26T01:05:50",
  "5590-01-29T01:04:28",
  "9999-10-24T01:26:12",
  "9999-02-28T01:51:32",
  "9999-12-03T23:59:59",
  "8651-11-30T03:22:27",
  "9194-12-31T05:54:33",
  "6338-12-31T07:15:41",
  "0735-02-10T23:59:59",
  "9999-01-31T22:55:04",
  "0688-12-26T05:31:01",
  "9999-12-31T08:42:41",
  "9999-03-14T21:47:25",
  "4494-11-30T23:59:59",
  "1253-03-31T23:59:59",
  "8016-12-12T10:54:12",
  "9999-12-16T22:56:24",
  "9999-02-28T23:59:59",
  "9999-12-16T13:40:49",
  "9999-06-08T11:21:39"
]
